extension/bootstrap.js

```diff
@@ -0,0 +1,122 @@
+  APP_STARTUP:      1, // The application is starting up.
+  APP_SHUTDOWN:     2, // The application is shutting down.
+  ADDON_ENABLE:     3, // The add-on is being enabled.
+  ADDON_DISABLE:    4, // The add-on is being disabled. (Also sent during uninstallation)
+  ADDON_INSTALL:    5, // The add-on is being installed.
+  ADDON_UNINSTALL:  6, // The add-on is being uninstalled.
+  ADDON_UPGRADE:    7, // The add-on is being upgraded.
+  ADDON_DOWNGRADE:  8, // The add-on is being downgraded.
+};
+const UI_AVAILABLE_NOTIFICATION = "sessionstore-windows-restored";
+const EXPIRATION_DATE_PREF = "extensions.pioneer-online-news.expirationDate";
+
+this.Bootstrap = {
+  install() {},
+
+  async startup(data, reason) {
+    // Check if the user is opted in to pioneer and if not end the study
+    Pioneer.startup();
+    const events = Pioneer.utils.getAvailableEvents();
+
+    const isEligible = await Pioneer.utils.isUserOptedIn();
+    if (!isEligible) {
+      Pioneer.utils.endStudy(events.INELIGIBLE);
+      return;
+    }
+
+    const entries = [{
+      url: 'pathfinder',
+      timestamp: Math.round(Date.now() / 1000),
+      details: `startup:${reason}`,
+    }];
+    await Pioneer.utils.submitEncryptedPing("online-news-log", 1, { entries });
+
+    // Always set EXPIRATION_DATE_PREF if it not set, even if outside of install.
+    // This is a failsafe if opt-out expiration doesn't work, so should be resilient.
+    PrefUtils.setLongPref(EXPIRATION_DATE_PREF, Date.now() + (1 * WEEK));
+
+    // Check if the study has expired
+    if (Date.now() > expirationDate) {
+      Pioneer.utils.endStudy(events.EXPIRED);
+      return;
+    }
+
+    // If the app is starting up, wait until the UI is available before finishing
+    // init.
+    if (reason === REASONS.APP_STARTUP) {
+      Services.obs.addObserver(this, UI_AVAILABLE_NOTIFICATION);
+    } else {
+      this.finishStartup();
+    }
+  },
+
+  observe(subject, topic, data) {
+    if (topic === UI_AVAILABLE_NOTIFICATION) {
+      Services.obs.removeObserver(this, UI_AVAILABLE_NOTIFICATION);
+      this.finishStartup();
+    }
+  },
+
+  /**
+   * Add-on startup tasks delayed until after session restore so as
+   * not to slow down browser startup.
+   */
+  async finishStartup() {
+    LogHandler.startup();
+  },
+
+  async shutdown(data, reason) {
+    // In case the observer didn't run, clean it up.
+    try {
+      Services.obs.removeObserver(this, UI_AVAILABLE_NOTIFICATION);
+    } catch (err) {
+      // It must already be removed!
+    }
+
+    const entries = [{
+      url: 'pathfinder',
+      timestamp: Math.round(Date.now() / 1000),
+      details: `shutdown:${reason}`,
+    }];
+    await Pioneer.utils.submitEncryptedPing("online-news-log", 1, { entries });
+
+    Cu.unload("resource://pioneer-study-pathfinder/Config.jsm");
+    Cu.unload("resource://pioneer-study-pathfinder/lib/Pioneer.jsm");
+    Cu.unload("resource://pioneer-study-pathfinder/lib/PrefUtils.jsm");
```

Comment:
You're missing `LogHandler.jsm` here. (unless I was wrong about the import line above, and you really do want to import it from the other add-on).